client/network/src/service.rs

```diff
@@ -332,7 +357,12 @@ impl<B: BlockT + 'static, H: ExHashT> NetworkWorker<B, H> {
 				transport::build_transport(local_identity, config_mem, config_wasm)
 			};
 			let mut builder = SwarmBuilder::new(transport, behaviour, local_peer_id.clone())
-				.peer_connection_limit(crate::MAX_CONNECTIONS_PER_PEER)
+				.connection_limits(ConnectionLimits::default()
+					.with_max_established_per_peer(Some(crate::MAX_CONNECTIONS_PER_PEER as u32))
+					// TODO: with_max_established_incoming(Some(<limit-from-somewhere>))
```

Comment:
Looking at the metrics of our Polkadot sentry nodes, none of those nodes have more than 300 incoming connections at a given point in time.

With that in mind, how about setting the limit already today, just not to a very strict value, e.g. `10_000`.

In the future we can then make this value configurable, setting a lower default value, allowing e.g. bootnodes to increase the limit manually.

What do you think?

Reply:
Sounds fine to me, I just didn't know what value to take and figured it may need to be configurable, but I didn't want to bloat this PR with new CLI configuration options or similar things. So you think `10_000` would be a good start? Any other suggestions or opposing opinions?

Reply:
`10_000` looks fine to me. The goal is to resist a little bit to a DoS attack. If you increase your ulimit to, say, one million, you don't want an attacker to occupy one million TCP connections.

Reply:
Done in https://github.com/paritytech/substrate/pull/7606/commits/4933c2ffd83483cb00070b90c624d6f6711a0bb9.